grails-app/domain/org/pih/warehouse/order/Order.groovy

```diff
@@ -518,6 +518,11 @@ class Order implements Serializable {
         return orderType?.isTransferOrder()
     }
 
+    Boolean isFullyInvoiceable() {
+        Set<OrderItem> encumberedOrderItems = orderItems.findAll{ it -> it.encumbered }
+        return encumberedOrderItems && !encumberedOrderItems.any{ it -> !it.invoiceable }
```

Comment:
Would not `encumberedOrderItems?.every { it -> it.invoiceable }` be cleaner? Plus this should also look into orderAdjustments, because if you won't have order items, just order adjustments, then you will always get the warning. On top of that `invoiceable` checks if item is invoiceable, not fully invoiceable, as you can see, there is 
`quantityAvailableToInvoice > 0` and in you case you are more interested in if `quantityAvailableToInvoice == quantity remaining to invoice`

Reply:
good point on the adjustments but as for `return encumberedOrderItems?.every { it -> it.invoiceable }` this will not work since it is not the same as  `return encumberedOrderItems && encumberedOrderItems.every { it -> it.invoiceable }`
The first check for `encumberedOrderItems` is for a case where we do `.every` on an empty list that will return `true` and we don't want that